Refactored code based on Issue #2

TcBello opened the pull request:
1. Added service layer on other authentication methods
2. Added docblocs
3. Added validation rules for AuthController and TodoController

Issue #2

TcBello committed:
ref: controllers, repositories and validators

Refactored AuthController, TodoController, AuthService, and Request Validators

mikco-hipe commented on app/Http/Controllers/Api/v1/AuthController.php:
```diff
@@ -22,18 +22,20 @@ public function index()
     }
-    public function login(Request $request)
+
+    /**
+     * Login method for user
+     * @param \App\Http\Requests\Auth\LoginUserRequest $request
+     */
```
no need for `@param`

mikco-hipe commented on app/Http/Controllers/Api/v1/AuthController.php:
```diff
@@ -22,18 +22,20 @@ public function index()
+     * Login method for user
+     * @param \App\Http\Requests\Auth\LoginUserRequest $request
+     */
+    public function login(LoginUserRequest $request)
```
add return type (check if it returns `JsonResponse`

```suggestion
    public function login(LoginUserRequest $request): JsonResponse
```

mikco-hipe commented on app/Http/Controllers/Api/v1/AuthController.php:
```diff
@@ -22,18 +22,20 @@ public function index()
-            if ($this->authService->attempt($validatedData)) {
-                $existingUser = Auth::user();
+            if ($this->authService->attempt($requestBody)) {
+                $existingUser = $this->authService->user();
```
This is good!

mikco-hipe reviewed (commented):
Please check!

TcBello committed:
refactor: controllers
Removed params on docblocs, added return types, added status on json response, added try catch on controllers

mikco-hipe commented on app/Http/Controllers/Api/v1/AuthController.php:
```diff
@@ -22,81 +23,133 @@ public function index()
     }
 
-    public function register(Request $request)
+    /**
+     * Register method for user
+     */
+    public function register(RegisterUserRequest $request): JsonResponse
```
yaaan tama! ganyan basis mo sa iba

mikco-hipe reviewed (approved):
LGTM
